cloud/vsphere/machineactuator.go

```diff
@@ -226,18 +227,22 @@ func (vc *VsphereClient) saveStartupScript(cluster *clusterv1.Cluster, machine *
 			},
 		)
 		if err != nil {
-			return err
+			return "", err
 		}
 	}
 
 	glog.Infof("Saving startup script for machine %s", machine.ObjectMeta.Name)
 
 	// Save the startup script.
-	if err := saveFile(startupScript, path.Join("/tmp", "machine-startup.sh"), 0644); err != nil {
-		return errors.New(fmt.Sprintf("Could not write startup script %s", err))
+	startupScriptPath, err := createStartupScriptPath("/tmp", startupScriptFilename)
```

Comment:
Instead of still saving the startup script in `/tmp/machine-startup.sh`, what if you saved them to the machine's staging dir (which already exists at this point, and is guaranteed to be unique path per-machine)?

```
machinePath := fmt.Sprintf(MachinePathStageFormat, machineName)
```

Reply:
changed to use machinePath